src/sentry/ratelimits/sliding_windows.py

```diff
@@ -337,7 +337,18 @@ def check_within_quotas(
         # (https://stackoverflow.com/a/3812600/1544347), there are no formal
         # guarantees about it.
         ordered_keys_to_fetch = list(keys_to_fetch)
-        redis_results = dict(zip(ordered_keys_to_fetch, self.client.mget(ordered_keys_to_fetch)))
+
+        with self.client.pipeline(transaction=False) as pipeline:
+            for key in ordered_keys_to_fetch:
+                pipeline.get(key)
+
+            results = iter(pipeline.execute())
+
+            ordered_keys_results = []
+            for _ in range(len(ordered_keys_to_fetch)):
+                ordered_keys_results.append(next(results))
+
+            redis_results = dict(zip(ordered_keys_to_fetch, ordered_keys_results))
 
```

Comment:
this code works fine, but in this pipeline you can just write `dict(zip(ordered_keys_to_fetch, results))` since there is nothing coming after this command (on line 352, the `results` iterator is exhausted)

Reply:
Actually isn't the `redis_results` being used again later?

Reply:
redis_results is defined here because that's the only place you have the data to define it. it's not a problem that it's defined in the with-block, the with-block does not create a namespace of any sort